### PR TITLE
add globalThis.VatData.passStyleOf to liveslots-hosted vat environment

### PR DESCRIPTION
**Pull request by warner**
Currently staged on https://github.com/Agoric/agoric-sdk/pull/8051 

Change liveslots to add `passStyleOf` to `globalThis.VatData` in the environment provided to real vats.

This is added as a string-named property, rather than a symbol-named one, which might not be the current plan.

### CODE REVIEW

**Review by erights — CHANGES_REQUESTED** [review]
> This is added as a string-named property, rather than a symbol-named one, which might not be the current plan.

Yeah, this is not the current plan, which is part of why #8051 remained on ice. See 
https://github.com/endojs/endo/blob/10edfa0e63bff509e85084cbfbc5f7b255f24469/packages/pass-style/src/passStyleOf.js#L198-L216
```js
export const PassStyleOfEndowmentSymbol = Symbol.for('@endo passStyleOf');

/**
 * If there is already a PassStyleOfEndowmentSymbol property on the global,
 * then presumably it was endowed for us by liveslots with a `passStyleOf`
 * function, so we should use and export that one instead.
 * Other software may have left it for us here,
 * but it would require write access to our global, or the ability to
 * provide endowments to our global, both of which seems adequate as a test of
 * whether it is authorized to serve the same role as liveslots.
 *
 * NOTE HAZARD: This use by liveslots does rely on `passStyleOf` being
 * deterministic. If it is not, then in a liveslot-like virtualized
 * environment, it can be used to detect GC.
 *
 * @type {PassStyleOf}
 */
export const passStyleOf =
  (globalThis && globalThis[PassStyleOfEndowmentSymbol]) ||
```

Note that it looks up a symbol-named property on the global itself, not on `globalThis.VatData`. My original design as of #8051 had @endo/pass-style look for it in `VatData`. But this was rightly objected to as too weird a layering violation, so we moved it into a symbol-named property directly on the global.

Please add a test showing that @endo/pass-style is reexporting this endowment from liveslots, rather than creating its own. That test should fail if liveslots exports the endowment via `VatData`, or as a string-named property.

**Comment by mhofman** [comment]
I believe this will require a change to `@endo/bundle-source` to have `importBundle` use `Reflect.ownKeys` instead of `Object.keys` to enumerate the `inescapableGlobalProperties`.

**Comment by warner** [comment]
I pushed a commit which changes liveslots to use the symbol-named property, and update the tests. The swingset test (which uses `importBundle()`) fails, as @mhofman predicted.

Since `import-bundle` is a package that lives in the Endo repo, I'm not sure how best to do the next step. Maybe `patch-package` in agoric-sdk to modify its version of `@endo/import-bundle` to respect symbol-named properties? The final PR needs to use on a newer version of Endo, so there's a necessary intermediate step of an Endo PR and subsequent release.

**Comment by mhofman** [comment]
> I'm not sure how best to do the next step

What we've done before in these cases is to start with an endo PR. Once approved and merged, either wait for the next version of endo to be released and picked up, or if "urgent", apply the endo change as a patch as you mention. I do recommend that during the endo PR process, the solution is verified through a patch or simulated upgrade of endo in agoric-sdk (`#endo-branch: endo-branch-name` in PR description).

File: packages/swingset-liveslots/src/liveslots.js [hunk 1]
```diff
@@ -1338,7 +1335,9 @@ function build(
       makeScalarBigSetStore: collectionManager.makeScalarBigSetStore,
       makeScalarBigWeakSetStore: collectionManager.makeScalarBigWeakSetStore,
     },
-  });
+    [PassStyleOfEndowmentSymbol]: passStyleOf,
+  };
+  harden(vatGlobals);
```
Comment (erights) [inline packages/swingset-liveslots/src/liveslots.js]: No objection to doing the `harden` here. But curious about your motivation for moving it? There's no mutation between the initialization and the harden, so the more traditional harden around the object literal should still be equivalent.

**Comment by erights** [comment]
@warner I assigned this to you. Is this right?

File: packages/swingset-liveslots/src/liveslots.js (hunk 1 above)
Comment (warner) [reply packages/swingset-liveslots/src/liveslots.js]: I don't remember why I moved it.. I'll move it back.

**Review by erights — CHANGES_REQUESTED** [review]
@warner 
Subsumed by https://github.com/Agoric/agoric-sdk/pull/9874 ? 
Should this be closed in favor of https://github.com/Agoric/agoric-sdk/pull/9874 ?

**Comment by warner** [comment]
yep

closing, subsumed by #9874 , which landed a few weeks ago

**Pull request closed without merging**